tests/nonlinear_channel_flow_Newton_Stokes/screen-output

```diff
@@ -0,0 +1,171 @@
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 147+0 iterations.
+      Total relative residual after nonlinear iteration 16: 3.39751e-05, norm of the rhs: 1.89797e+07, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 98+0 iterations.
+      Total relative residual after nonlinear iteration 17: 1.02507e-05, norm of the rhs: 7.04128e+06, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 86+0 iterations.
+      Total relative residual after nonlinear iteration 18: 3.80259e-06, norm of the rhs: 2.70395e+06, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 92+0 iterations.
+      Total relative residual after nonlinear iteration 19: 1.46026e-06, norm of the rhs: 1.05504e+06, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 85+0 iterations.
+      Total relative residual after nonlinear iteration 20: 5.69767e-07, norm of the rhs: 413589, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 91+0 iterations.
+      Total relative residual after nonlinear iteration 21: 2.23359e-07, norm of the rhs: 162464, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 91+0 iterations.
+      Total relative residual after nonlinear iteration 22: 8.77391e-08, norm of the rhs: 63925.5, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 98+0 iterations.
+      Total relative residual after nonlinear iteration 23: 3.45245e-08, norm of the rhs: 24729.4, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 95+0 iterations.
+      Total relative residual after nonlinear iteration 24: 1.33549e-08, norm of the rhs: 9750.95, newton_derivative_scaling_factor: 1
+
+   Skipping temperature solve because RHS is zero.
+   The linear solver tolerance is set to 0.1
```

Comment:
It seems the linear solver tolerance for this model is always set to the maximum value. Can you add a test that uses the tolerance scaling, and maybe also the newton derivative scaling method? Maybe just tweaking some parameters in this test would do the trick already. Otherwise we can not be sure the method still works when we are going to restructure the nonlinear solver scheme code (related to #1279).